site/docs/concepts/derivations.md

```diff
@@ -7,700 +7,769 @@ import Mermaid from '@theme/Mermaid';
+as well as those which are nested within a source document.
+Consider the following schematized document:
+```json
+{"top-level": true, "object": {"foo": 42}, "arr": ["bar"]}
 ```
 
-</TabItem>
-</Tabs>
+In your SQL code, you can use parameters like `$top_level`, `$object$foo`, or `$arr$0`.
+If you're unsure of what parameter to use for a given field,
+try typing something approximate and Flow will suggest the appropriate `$parameter`.
 
-Source transfers are read twice.
-The first read shuffles on `/recipient`
-to track account credits,
-and the second shuffles on `/sender`
-to track account debits and to publish enriched transfer events.
-Update lambdas return the amount of credit or debit,
-and these amounts are summed
-into a derivation register keyed on the account.
+### Migrations
 
-<Mermaid chart={`
-  sequenceDiagram
-    autonumber
-    Derivation->>Registers: lookup(key = alice)?
-    Registers-->>Derivation: not found, initialize as 0
-    Derivation->>Update λ: update({recipient: alice, amount: 50, ...})?
-    Update λ-->>Derivation: return +50
-    Derivation->>Update λ: update({sender: alice, amount: 75, ...})?
-    Update λ-->>Derivation: return -75
-    Derivation-->>Derivation: Register: 0 + 50 => 50
-    Derivation-->>Derivation: Register: 50 - 75 => -25
-    Derivation->>Publish λ: publish({sender: alice, amount: 75, ...}, register = -25, previous = 50)?
-    Publish λ-->>Derivation: return {sender: alice, amount: 75, balance: -25, overdrawn: true}
-`}/>
+The SQLite connector offers a managed, persistent SQLite database
+that can accommodate any number of tables, indices, views, triggers, and other schema,
+as defined by your database migrations.
+To add a migration, simply append it to the `migrations` array,
+either as a block of inline SQL statements
+or as a relative path to a file of SQL statements:
 
-### Creating TypeScript modules
+```yaml
+derive:
+  using:
+    sqlite:
+      migrations:
+        - CREATE TABLE foo (thing INTEGER NOT NULL);
+          CREATE INDEX idx_foo_thing foo (thing);
+        - ../path/to/another/migration.sql
+        - ALTER TABLE foo ADD COLUMN other_thing TEXT NOT NULL;
+        - https://example.com/yet/another/migration.sql
+```
+
+:::caution
+You cannot change an existing migration once it has been published.
+Instead, add a new migration which applies your desired schema.
+:::
 
-To create a new TypeScript module for the lambdas of your derivation,
-you can use `flowctl typescript generate` to generate it.
-In the derivation specification, choose the name for the new module and
-run `flowctl typescript generate`.
-Flow creates a module with the name you specified, stubs of the required interfaces,
-and TypeScript types that match your schemas.
-Update the module with your lambda function bodies,
-and proceed to test and deploy your catalog.
+Collectively, your migrations serve as the [internal state](#internal-state) of your derivation.
```

Comment:
On first read, I got a little hung up on the fact that these are called 'migrations.' I see how they're different than any old SQL queries, and they're also separate from the concept of internal state more generally (because they're not in TS derivations). But still, once I got down to the "Internal State" section, I was like: wait, I just got used to this new term, "Migration," and now it's not being used anymore. So a bit of cognitive friction. 

Unfortunately I don't have a solution to offer, but I wanted to throw this in anyway.

Reply:
I'd been using `migrations` as a kind-of short hand for "The SQLite database schema you create that is the internal state of your derivation". I took a pass and tried to fully unpack it throughout, explicitly connecting that the schema _created by_ migrations _is_ the internal task state.